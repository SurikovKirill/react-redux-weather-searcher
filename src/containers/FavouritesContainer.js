import React, {Component} from 'react';
import Search from '../components/Search';
import FavouritesList from '../components/FavouritesList';
import { connect } from 'react-redux';
import { addNewCityAsync, removeCity } from '../actions/favourites';

class FavouritesContainer extends Component {

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.addCity(e.target.cityName.value);
        e.target[0].value = '';
    };

    handleRemove = (name) => {
        this.props.removeCity(name);
    };

    render() {
        console.log("FavContainer");
        console.log(this.props.cities);
        console.log(this.props.isLoading);
        console.log(this.props.errors);
        return (
            <>
                <Search onSubmit={this.handleSubmit} />
                <FavouritesList cities={this.props.cities} onRemove={this.handleRemove} isLoading={this.props.isLoading} errors={this.props.errors}/>
            </>
        );
    }
}


FavouritesContainer.defaultProps = {
    cities: [],
    isLoading: [],
    errors: [],
    removeCity: f => f,
    addCity: f => f,
};

const mapStateToProps = ({ favouritesReducer: { cities, isLoading, errors } }) => ({
    cities,
    isLoading,
    errors,
});

const mapDispatchToProps = {
    addCity: addNewCityAsync,
    removeCity: removeCity,
};


export default connect(mapStateToProps, mapDispatchToProps)(FavouritesContainer);