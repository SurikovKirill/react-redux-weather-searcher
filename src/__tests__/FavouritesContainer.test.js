import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import FavouritesContainer from '../containers/FavouritesContainer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let city;
let cities;
beforeEach(() => {
    city = {
        isLoading: false,
        weather: {
            name: 'Helsinki',
            img: 'i',
            temperature: 't',
            wind: 'w',
            cloudiness: 'c',
            pressure: 'p',
            humidity: 'h',
            location: 'l'
        },
        error: false,
    };
    cities = [{
        name: 'Moscow',
        img: 'i1',
        temperature: 't1',
        wind: 'w1',
        cloudiness: 'c1',
        pressure: 'p1',
        humidity: 'h1',
        location: 'l1'
    }];
});

describe('FavouritesContainer container', () => {

    test('matches the snapshot when favorites cities are loaded', () => {
        const store = mockStore({
            cityReducer: city,
            favouritesReducer: {
                cities: cities,
                errors: [],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <FavouritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot when favorites cities are loading', () => {

        const store = mockStore({
            cityReducer: city,
            favouritesReducer: {
                cities: cities,
                errors: [],
                isLoading: ['Moscow']
            }
        });

        const tree = renderer.create(
            <Provider store={store}>
                <FavouritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot on error', () => {

         const store = mockStore({
            cityReducer: city,
            favouritesReducer: {
                cities: cities,
                errors: ['Moscow'],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <FavouritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});