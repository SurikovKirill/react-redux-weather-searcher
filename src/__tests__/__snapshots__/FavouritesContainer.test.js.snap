// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavouritesContainer container matches the snapshot on error 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      />
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavouriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="i1"
        />
        <h1>
          t1
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="Error"
      >
        <h1>
          Не удалось загрузить данные
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`FavouritesContainer container matches the snapshot when favorites cities are loaded 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      />
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavouriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="i1"
        />
        <h1>
          t1
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="WeatherInfo"
      >
        <div
          className="Info"
        >
          <h3>
            Ветер
          </h3>
          <h4>
            w1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Облачность
          </h3>
          <h4>
            c1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Давление
          </h3>
          <h4>
            p1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Влажность
          </h3>
          <h4>
            h1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Координаты
          </h3>
          <h4>
            l1
          </h4>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`FavouritesContainer container matches the snapshot when favorites cities are loading 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      />
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavouriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="i1"
        />
        <h1>
          t1
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="Loader"
      >
        <h1>
          Подождите, данные загружаются
        </h1>
        <div
          className="Spinner"
        />
      </div>
    </div>
  </div>,
]
`;
